test/units/testsuite-04.sh

```diff
@@ -141,4 +141,87 @@ sleep 3
 # https://github.com/systemd/systemd/issues/15528
 journalctl --follow --file=/var/log/journal/*/* | head -n1 || [[ $? -eq 1 ]]
 
+function add_logs_filtering_override() {
+    UNIT=$1
+    OVERRIDE_NAME=$2
+    LOG_FILTER=${3:-""}
+
+    mkdir -p /etc/systemd/system/"$UNIT".d/
+    echo "[Service]" > /etc/systemd/system/logs-filtering.service.d/"${OVERRIDE_NAME}".conf
+    echo "LogFilterPatterns=$LOG_FILTER" >> /etc/systemd/system/logs-filtering.service.d/"${OVERRIDE_NAME}".conf
+    systemctl daemon-reload
+}
+
+function run_service_and_fetch_logs() {
+    UNIT=$1
+
+    START=$(date '+%Y-%m-%d %T.%6N')
+    systemctl restart "$UNIT"
+    sleep .5
+    journalctl --sync
+    END=$(date '+%Y-%m-%d %T.%6N')
+
+    journalctl -q -u "$UNIT" -S "$START" -U "$END" | grep -Pv "systemd\[[0-9]+\]"
+    systemctl stop "$UNIT"
+}
+
+function is_xattr_supported() {
+    START=$(date '+%Y-%m-%d %T.%6N')
+    systemd-run --unit text_xattr --property LogFilterPatterns=log sh -c "sleep .5"
+    sleep .5
+    journalctl --sync
+    END=$(date '+%Y-%m-%d %T.%6N')
+    systemctl stop text_xattr
+
+    LOGS=$(journalctl -q -u "text_xattr" -S "$START" -U "$END")
+    FAILED_MSG="Failed to set 'user.journald_log_filter_patterns' xattr"
+    NOTSUP_MSG="not supported"
+    if [[ "$LOGS" == *"$FAILED_MSG"* && "$LOGS" == *"$NOTSUP_MSG"* ]]; then
+        echo 0
+    else
+        echo 1
+    fi
+}
+
+if [[ $(is_xattr_supported) -eq 1 ]]; then
```

Comment:
That seems unnecessary. Just make `is_xattr_supported` return true or false, and do
```bash
if is_xattr_supported; then
     …
```